classes/journal/JournalDAO.inc.php

```diff
@@ -418,6 +422,67 @@ function resequenceJournals() {
 	function getInsertJournalId() {
 		return $this->getInsertId('journals', 'journal_id');
 	}
+	
+	/**
+	 * Get journals by setting.  Backported from master pkp-lib's ContextDAO.
+	 * @param $settingName string
+	 * @param $settingValue mixed
+	 * @param $contextId int
+	 * @return DAOResultFactory
+	 */
+	function getBySetting($settingName, $settingValue, $contextId = null) {
+		$params = array($settingName, $settingValue);
+		if ($contextId) $params[] = $contextId;
+
+		$result = $this->retrieve(
+			'SELECT * FROM ' . $this->_getTableName() . ' AS c
+			LEFT JOIN ' . $this->_getSettingsTableName() . ' AS cs
+			ON c.' . $this->_getPrimaryKeyColumn() . ' = cs.' . $this->_getPrimaryKeyColumn() .
+			' WHERE cs.setting_name = ? AND cs.setting_value = ?' .
```

Comment:
May as well use `INNER JOIN` rather than `LEFT JOIN`. An `a LEFT JOIN b ... WHERE b.something = ?` is equivalent to the same with `INNER JOIN`.